### DISCUSSION TITLE
Generate mermaid diagrams out of comment blocks within C++ code.

### DISCUSSION THREAD

**Discussion by VintageDesign** [mermaid-js/mermaid#3502] [Q&A]
I realize this isn't directly related to Mermaid but does anyone know of a C++ documentation tool that is capable of parsing mermaid diagrams and rendering them?

**Comment by emersonbottero** [comment]
You can use [Vitepress](https://vitepress.vuejs.org/) and use the [Mermaid Plugin](https://emersonbottero.github.io/vitepress-plugin-mermaid/) for it.
 you can use together with any other language.. 
 I use to document C# libraries for example.

**Comment by aloisklink** [comment]
You can also setup `doxygen` to render Mermaid files too https://github.com/tttapa/doxygen-mermaid.

**Comment by VintageDesign** [comment]
Nice thanks!
